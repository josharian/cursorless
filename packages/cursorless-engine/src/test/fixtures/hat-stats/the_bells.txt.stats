nTokens: 782
nHats:
	mean: 99%
	std: 0%
	min: 99%
	max: 99%
	spark: 0              25               50               75            █  100

nPenalty0:
	mean: 5%
	std: 0%
	min: 5%
	max: 5%
	spark: 0   █          25               50               75               100

nPenalty1:
	mean: 54%
	std: 0%
	min: 53%
	max: 55%
	spark: 0              25               50  ▄█           75               100

nPenalty2:
	mean: 40%
	std: 0%
	min: 40%
	max: 41%
	spark: 0              25        ▅█     50               75               100

