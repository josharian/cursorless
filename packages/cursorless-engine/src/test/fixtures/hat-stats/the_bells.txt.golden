*   *
The Bells
[-] [---]

 *  *        * *   *    *   *  * *       *   *
by Edgar Allan Poe 1809 – 1849 (public domain)
[] [---] [---] [-] [--] # [--] #[----] [----]#

**
I.
##

                                x      x
        *      * *       *      *   *  *
        Hear the sledges with the bells—
        [--] [-] [-----] [--] [-] [---]#

                        x
                    *   *    *
                 Silver bells!
                 [----] [---]#

     x                     x         x         xx
  *  *  *     *   *        *         *         **
What a world of merriment their melody foretells!
[--] # [---] [] [-------] [---] [----] [-------]#

          x    v         x     x v     v
          *    *    *  * *     * *     *
        How they tinkle, tinkle, tinkle,
        [-] [--] [----]# [----]# [----]#

           x   v      v    x      v
           *   *   *  *    *   *  *
           In the icy air of night!
           [] [-] [-] [-] [] [---]#

        v     w   v      w        x
        *     *   *      *        *
        While the stars that oversprinkle
        [---] [-] [---] [--] [----------]

        w   o      x   w    x  x  w
        *   *      *   *    *  *  *
        All the heavens, seem to twinkle
        [-] [-] [-----]# [--] [] [-----]

              o o x           x
              * * *           *      *
           With a crystalline delight;
           [--] # [---------] [-----]#

            v      v o #   # ^   ^
            *      * * *   * *   *
         Keeping time, time, time,
         [-----] [--]# [--]# [--]#

          x #  v    v  x      w  0
          * *  *    *  *      *  *
         In a sort of Runic rhyme,
         [] # [--] [] [---] [---]#

0   #         v         ^   w      v     o
*   *         *         *   *      *     *
To the tintinabulation that so musically wells
[] [-] [-------------] [--] [] [-------] [---]

       w    >   w    > o    ~ #    + ^
       *    *   *    * *    * *    * *    b
       From the bells, bells, bells, bells,
       [--] [-] [---]# [---]# [---]# [---]#

               0      >      ~    v
               *    g *    r *    *
               Bells, bells, bells—
               [---]# [---]# [---]#

  o    ~              v +      x     w   0  +    x
  *    *   *          * *      *     *   *  *    *
  From the jingling and the tinkling of the bells.
  [--] [-] [------] [-] [-] [------] [] [-] [---]#

v v
* *
II.
[]#

           x          #   w
           * b        *   *     b    p
        Hear the mellow wedding bells,
        [--] [-] [----] [-----] [---]#

                  o          w
                  *     g    *
                 Golden bells!
                 [----] [---]#

  ^  0   v    #   w                   o       x  o
  *  *   *    *   *       g           *       *  *
What a world of happiness their harmony foretells!
[--] # [---] [] [-------] [---] [-----] [-------]#

            v     v        w   ^ v
            *     * r      *   * *
        Through the balmy air of night
        [-----] [-] [---] [-] [] [---]

          ^   w  w     w   >    o      #
          *   *  *     *   *    *      *
        How they ring out their delight!
        [-] [--] [--] [-] [---] [-----]#

              w          w x          o
              * r        ***          *y
           From the molten-golden notes,
           [--] [-] [----]#[----] [---]#

               >    v  o   o
               *    *  *   *  1
               And all in tune,
               [-] [-] [] [--]#

           0    ~            # 0
           *    *   *        * *
           What a liquid ditty floats
           [--] # [----] [---] [----]

         o   o   x  v    +       o   >     #   v
    p    *   *   *  *    *       * 2 *     *   *
    To the turtle-dove that listens, while she gloats
    [] [-] [----]#[--] [--] [-----]# [---] [-] [----]

                #  ~  o   ^
                *  *  *   *
               On the moon!
               [] [-] [--]#

          +x >     #    #     #    w    v
          *b *     *    *     *    *    b
         Oh, from out the sounding cells,
         []# [--] [-] [-] [------] [---]#

~      w     ~   o     w            +    0
*    b *     *   *     *            *    *
What a gush of euphony voluminously wells!
[--] # [--] [] [-----] [----------] [---]#

                      ^     >
               b    y *     *
               How it swells!
               [-] [] [----]#

                   #  ^
               g   *  *
               How it dwells
               [-] [] [----]

            ^         # ~     ^    w
            *  r      * *   b *    *
           On the Future! how it tells
           [] [-] [----]# [-] [] [---]

            +                   #
            *  p   g       y    *
           Of the rapture that impels
           [] [-] [-----] [--] [----]

          #         o      0     ^
          *  1      *      * 1   *
         To the swinging and the ringing
         [] [-] [------] [-] [-] [-----]

                       w      o      #
            b 2   p    b y    b 1    b
           Of the bells, bells, bells,
           [] [-] [---]# [---]# [---]#

         ^           ^  ^   0  0   >  >   ~
         *   2  2    b  *   b  *   b  *   b
         Of the bells, bells, bells, bells,
         [] [-] [---]# [---]# [---]# [---]#

                ~   +  +   x      w
                *   b  *   g  b   *
               Bells, bells, bells—
               [---]# [---]# [---]#

   0       ^             o                o  +
   *   g   *     r     r *        g   p   *  *
  To the rhyming and the chiming of the bells!
  [] [-] [-----] [-] [-] [-----] [] [-] [---]#

0  w
*  *
III.
[-]#

            0       ^       ^     0o
            *   y   *       *     **
         Hear the loud alarum bells—
         [--] [-] [--] [----] [---]#

                          #
                    *     *  b
                 Brazen bells!
                 [----] [---]#

       ^       >   v 0  w    >           0     >
g      *   r   *   g *  g    *           *     *g
What tale of terror, now, their turbulency tells!
[--] [--] [] [----]# [-]# [---] [--------] [---]#

       ~             >          #
       *    1        * 2    p   *
       In the startled ear of night
       [] [-] [------] [-] [] [---]

              >      0  0     +
         r    *      *  *     *   y      r
       How they scream out their affright!
       [-] [--] [----] [-] [---] [------]#

          >  >            ~  ~     vo
          *  *            *  *     *g
         Too much horrified to speak,
         [-] [--] [-------] [] [---]#

            ~   >   0    ~   # ~     ^
            *   *   *    *   g *     g
         They can only shriek, shriek,
         [--] [-] [--] [----]# [----]#

                   >       ~ 0
                   *   1   * g
                  Out of tune,
                  [-] [] [--]#

        ~              ^  +  x      +    x       >
b  p    *              *  *  b      *  2 b    g  g
In a clamorous appealing to the mercy of the fire,
[] # [-------] [-------] [] [-] [---] [] [-] [--]#

 +     +                      x                #     ~
 * y   *  *            p      b b     b        *  r  g
In a mad expostulation with the deaf and frantic fire,
[] # [-] [-----------] [--] [-] [--] [-] [-----] [--]#

                  0      ++       x       v
                  *      *g  p    r  y    r
            Leaping higher, higher, higher,
            [-----] [----]# [----]# [----]#

                     +             w
            y    1   *          1  r
            With a desperate desire,
            [--] # [-------] [----]#

                             o
          g  2    b          *
         And a resolute endeavor
         [-] # [------] [------]

            #                #  o
           1*r    g  2   b   *  r
         Now—now to sit or never,
         [-]#[-] [] [-] [] [---]#

           v          v   #   v  ^   +   o
        b  b    g  r  b   *   *  *   *   *
       By the side of the pale-faced moon.
       [] [-] [--] [] [-] [--]#[---] [--]#

              # w        ^   >  0   ~
            p r b       br   *  r   *  p
            Oh, the bells, bells, bells!
            []# [-] [---]# [---]# [---]#

                 x   +
            2    b   *      g     y      g
            What a tale their terror tells
            [--] # [--] [---] [----] [---]

                        ^
                  1     *   y
                  Of Despair!
                  [] [-----]#

                    >>          ~
        2     g     *r   r    r r  p  r   1
       How they clang, and clash, and roar!
       [-] [--] [---]# [-] [---]# [-] [--]#

       x    v              ~
       b    b   p       r  *
       What a horror they outpour
       [--] # [----] [--] [-----]

    w         x o             ~
 y  b      b  b b             *   y2
On the bosom of the palpitating air!
[] [-] [---] [] [-] [---------] [-]#

            o      x   +    w    +
       p    b    1 b   *    *    r
       Yet the ear it fully knows,
       [-] [-] [-] [] [---] [---]#

                #      +   x
             y  b      *   p
            By the twanging,
            [] [-] [------]#

                 ^  0       v
              p  b  *       p
            And the clanging,
            [-] [-] [------]#

          x  #                        x
          b  b        2    p  1   b   *
         How the danger ebbs and flows;
         [-] [-] [----] [--] [-] [---]#

            0    x       >         w
       1    b    b       *      g  p
       Yet the ear distinctly tells,
       [-] [-] [-] [--------] [---]#

                >  x       o
             2  b  *       p
            In the jangling,
            [] [-] [------]#

                ^            #
              y b       b    *
            And the wrangling.
            [-] [-] [-------]#

        v  0             o            #
        b  b   1         *    2    r  p
       How the danger sinks and swells,
       [-] [-] [----] [---] [-] [----]#

   >           w  ~            v  +          v x        ^
 2 b         g b  b      p     b  b     r    b g     y  *
By the sinking or the swelling in the anger of the bells—
[] [-] [-----] [] [-] [------] [] [-] [---] [] [-] [---]#

             o  v        0
             b  g     1  *
             Of the bells—
             [] [-] [---]#

     #  w        ^      0      >      ~
     b  g     2  p     yp     1p     2p
     Of the bells, bells, bells, bells,
     [] [-] [---]# [---]# [---]# [---]#

            x    + v    x w    >
            b    p b    y b    *
            Bells, bells, bells—
            [---]# [---]# [---]#

 w  o            x #            w ^   o    x
 b  g      g     b g       p    b g   b    b
 In the clamor and the clangor of the bells!
 [] [-] [----] [-] [-] [-----] [] [-] [---]#

 ^^
 **
IV.
[]#

            w  0            o  ~  #    ~
            b  g         y  b  b  b    *
          Hear the tolling of the bells—
          [--] [-] [-----] [] [-] [---]#

                  v   ^    v
                  b   b    b
                 Iron bells!
                 [--] [---]#

v    o     v  #                   o            0   w
b    b     b  b     r     b       b  p         *   b
What a world of solemn thought their monody compels!
[--] # [---] [] [----] [-----] [---] [----] [-----]#

        #  >        ~   ^ ~        v
        b  g        *   b g     1  y
        In the silence of the night,
        [] [-] [-----] [] [-] [---]#

          w  v    0     +
          b  b    *     g       2
        How we shiver with affright
        [-] [] [----] [--] [------]

  #    w                    0    ^    x o
  b    b y          1       b    b    b b
  At the melancholy menace of their tone!
  [] [-] [--------] [----] [] [---] [--]#

        >    >            ^   x
        b    *      g     b   b
        For every sound that floats
        [-] [---] [---] [--] [----]

               o      o        #       0
           2   b  r   b        b       b
        From the rust within their throats
        [--] [-] [--] [----] [---] [-----]

                 0  > x    0
                 b  b b    *
                 Is a groan.
                 [] # [---]#

          w   ^ >     +~ w x   ~
          b   b *     *b y r   *     b
        And the people—ah, the people—
        [-] [-] [----]#[]# [-] [----]#

          x   +   #       >  v       +  o
          b   b   b    p  b  r       *  y
       They that dwell up in the steeple,
       [--] [--] [---] [] [] [-] [-----]#

                 x     ^  #
                 g     b  y
                 All alone,
                 [-] [---]#

          o ^         v^ w      0 o      >
          b b         by r      y r      y
        And who tolling, tolling, tolling,
        [-] [-] [-----]# [-----]# [-----]#

          ~    v           0      ~
          b    g   y       b      y
          In that muffled monotone,
          [] [--] [-----] [------]#

         ~    w    w  +  x  o
         b    g    b  b  b  b
         Feel a glory in so rolling
         [--] # [---] [] [] [-----]

          >   +           #  o v
          b   b   1       b  g b    g
          On the human heart a stone—
          [] [-] [---] [---] # [---]#

        v  ^    x     x    ~  0
        b  b    g     b    b  b    r
     They are neither man nor woman—
     [--] [-] [-----] [-] [-] [---]#

        w  0    v            +    v
        b  b    g       2    b    b  p
     They are neither brute nor human—
     [--] [-] [-----] [---] [-] [---]#

                 o  >  w
                 b  b  b     *
              They are Ghouls:
              [--] [-] [----]#

          #    w  #     #  w >     v  v
          b    g  *     r  b b     b  *
        And their king it is who tolls;
        [-] [---] [--] [] [] [-] [---]#

        #    0 ~    + +    x x    v
        g    b b    y b    1 g    1
        And he rolls, rolls, rolls,
        [-] [] [---]# [---]# [---]#

                    v
                    g
                    Rolls
                    [---]

             ^      +    ^   0    #
             g b    b    r   b    b
             A pæan from the bells!
             # [--] [--] [-] [---]#

            ^  o      #  x     ~
            b  g      b  g     b
          And his merry bosom swells
          [-] [-] [---] [---] [----]

               0    >       x >   >    ^
               r    b g     g r   b    b
             With the pæan of the bells!
             [--] [-] [--] [] [-] [---]#

            0 x     +  w 0   v  ^    w
            b g     *  1 g   g  b    *
          And he dances, and he yells;
          [-] [] [----]# [-] [] [---]#

          ^         w o ~   # +   ^
          *         b 1 r   1 r   1
          Keeping time, time, time,
          [-----] [--]# [--]# [--]#

           v >   w   v        w   0
           b g   g   g     b  g   1
          In a sort of Runic rhyme,
          [] # [--] [] [---] [---]#

             x   o        w v   ~
             p   g   y    g p   b    1
             To the pæan of the bells—
             [] [-] [--] [] [-] [---]#

               v  w   +    x
               g  p   b    *
               Of the bells:
               [] [-] [---]#

          0         o > o   ~ #   +
          *         b 1 p   1 p   1
          Keeping time, time, time,
          [-----] [--]# [--]# [--]#

           w ~   o   o         0  x
           b g   g   g     g   b  2
          In a sort of Runic rhyme,
          [] # [--] [] [---] [---]#

            ^   #      x      #   ~   w
            p   g      g      g   b   b  2
            To the throbbing of the bells—
            [] [-] [-------] [] [-] [---]#

          w  0   v    v w    w o    x
          g  p   g    2 g    2 g    b
          Of the bells, bells, bells—
          [] [-] [---]# [---]# [---]#

             o  ^      #    ^ >     o  o
             g  g      g    g p     b  *
            To the sobbing of the bells;
            [] [-] [-----] [] [-] [---]#

          >         # o ~   # +   ^
          *         b 2 p   2 p   2
          Keeping time, time, time,
          [-----] [--]# [--]# [--]#

            +  0  ~     0 +     >       ~
            g  g  *     2 *     2 b     2
            As he knells, knells, knells,
            [] [] [----]# [----]# [----]#

          ^  x                ^ +
          g  r   r       r    b 2
          In a happy Runic rhyme,
          [] # [---] [---] [---]#

             #  >        o  0 x   #    v
             g  g        b  g y   g    b
            To the rolling of the bells—
            [] [-] [-----] [] [-] [---]#

          ^  v   ^      0      >    w
          g  y   g      g      g    b
          Of the bells, bells, bells—
          [] [-] [---]  [---]  [---]#

             0  ~      0    >  +  ~
             g  g      g    g  g  g
            To the tolling of the bells,
            [] [-] [-----] [] [-] [---]

      >  w   +      x      v      w    o
      g  y   g      r      r      r    b
      Of the bells, bells, bells, bells—
      [] [-] [---]  [---]  [---]  [---]#

              o      #      ^    #
              r      r      r    b
              Bells, bells, bells—
              [---]  [---]  [---]#

   ~  x    v       > o    #        ~ #   0    >
   g  r    r       b y    g        g y   r    *
  To the moaning and the groaning of the bells.
  [] [-] [-----] [-] [-] [------] [] [-] [---]#

